gst-plugin/src/drpai-models/drpai-ssdv3/ssdv3_post_processor.cpp

```diff
@@ -0,0 +1,173 @@
+            const float ph  = priors_flat_.at(prior_idx + 3);
+
+            const float dx = inference_output_buf.at(boxes_start_index + (item*4) + 0);
+            const float dy = inference_output_buf.at(boxes_start_index + (item*4) + 1);
+            const float dw = inference_output_buf.at(boxes_start_index + (item*4) + 2);
+            const float dh = inference_output_buf.at(boxes_start_index + (item*4) + 3);
+
+            const float x = (pcx + dx * pw / loc_scale_xy) * static_cast<float>(img_width);
+            const float y = (pcy + dy * ph / loc_scale_xy) * static_cast<float>(img_height);
+            const float w = (pw * std::exp(dw / loc_scale_wh)) * static_cast<float>(img_width);
+            const float h = (ph * std::exp(dh / loc_scale_wh)) * static_cast<float>(img_height);
+
+            detections.emplace_back(Box{x, y, w, h}, pred_class, *max_pred);
+        }
+    }
+}
+
+void SSDV3_PostProcessor::open_resource(uint32_t inference_output_size,
+                                        uint32_t img_width,
+                                        uint32_t img_height,
+                                        uint32_t num_classes)
+{
+    // cache img_w/h and class count in base
+    BasePostProcessor::open_resource(inference_output_size, img_width, img_height, num_classes);
+
+    // model layout: N * (4 + num_classes)
+    const uint32_t stride = 4 + num_classes;
+    if (inference_output_size == 0 || (inference_output_size % stride) != 0) {
+        throw std::runtime_error("[ERROR][SSDV3] Output size is not divisible by (4 + num_classes).");
+    }
+    const uint32_t expected_count = inference_output_size / stride;
+
+    // read path to priors file from params
+    std::string priors_file = get_param("[priors_file]");
+    if (priors_file.empty()) {
+        throw std::runtime_error("[ERROR][SSDV3] Param [priors_file] leads to empty file.");
+    }
+
+    if (priors_file.front() != '/') {
+        priors_file = prefix + "/" + priors_file;
+    }
+
+    // load priors
+    load_priors_file(priors_file);
+
+    // sanity check file rows vs model expectations
+    priors_count_ = static_cast<uint32_t>(priors_flat_.size() / 4);
+    if (priors_count_ != expected_count) {
+        throw std::runtime_error("[ERROR][SSDV3] Priors count mismatch. File has " + std::to_string(priors_count_) +
+                                 ", model expects " + std::to_string(expected_count) + ".");
+    }
+
+    std::cout << "[INFO][SSDV3] Loaded priors: " << priors_count_
+              << " | loc_scale_xy=" << loc_scale_xy
+              << " loc_scale_wh=" << loc_scale_wh << std::endl;
+}
+
+void SSDV3_PostProcessor::load_priors_file(const std::string &path)
+{
+    std::ifstream in(path);
+    if (!in.is_open()) {
+        throw std::runtime_error("[ERROR][SSDV3] Failed to open priors file: " + path);
+    }
+
+    priors_flat_.clear();
+
+    std::string line;
+    uint64_t line_no = 0;
+
+    while (std::getline(in, line)) {
```

Comment:
No need to do line and space handling as `std::ifstream` does that for you. See how #66 is doing it.

So please remove `line_no`, `std::getline`, `find_first_not_of`, `std::istreangstream ss`, and `ss.peek`.